Update cargo

weihanglo opened the pull request:
15 commits in 17b3d0de0897e1c6b8ca347bd39f850bb0a5b9f6..9d5b32f503fc099c4064298465add14d4bce11e6
2023-02-17 19:45:09 +0000 to 2023-02-22 23:04:16 +0000

- refactor(job_queue): docs and move types around (rust-lang/cargo#11758)
- Scrub more of the test environment (rust-lang/cargo#11757)
- Make more reads of environment variables go through the `Config` (rust-lang/cargo#11754)
- Revert "Update curl-sys to use libcurl 7.88.1" (rust-lang/cargo#11755)
- use consistent case (rust-lang/cargo#11748)
- Switch some tests from `build` to `check` (rust-lang/cargo#11725)
- Fix typo in sparse-registry warning message (rust-lang/cargo#11753)
- reuse url encoding from `url` crate, don't use separate `percent-encoding` (rust-lang/cargo#11750)
- Read environment variables through `Config` instead of `std::env::var(_os)` (rust-lang/cargo#11727)
- Update curl-sys to use libcurl 7.88.1 (rust-lang/cargo#11749)
- mdman: update pretty_assertions to reduce deps (rust-lang/cargo#11747)
- Cleanup tests (rust-lang/cargo#11745)
- Enhance help texts of position args (rust-lang/cargo#11740)
- Fix typo (rust-lang/cargo#11741)
- Update comment about cargo-ok (rust-lang/cargo#11724)

weihanglo commented:
Lockfile change looks reasonable and expected.

@bors r+ p=1

bors commented:
:pushpin: Commit 391a243d6b3695631cd8f333b5ea1f9fd1b296d9 has been approved by `weihanglo`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=weihanglo 391a243d6b3695631cd8f333b5ea1f9fd1b296d9 -->
<!-- homu: {"type":"Approved","sha":"391a243d6b3695631cd8f333b5ea1f9fd1b296d9","approver":"weihanglo","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

akiirui commented:
I think we should wait for curl to release https://github.com/curl/curl/commit/821f6e2a89de8aec1c7da3c0f381b92b2b801efc, and update curl-sys to prevent https://github.com/rust-lang/cargo/issues/11746.

bors commented:
:hourglass: Testing commit 391a243d6b3695631cd8f333b5ea1f9fd1b296d9 with merge a5493c220346b834c479119a0a356717fd6c1c3e...
<!-- homu: {"type":"BuildStarted","head_sha":"391a243d6b3695631cd8f333b5ea1f9fd1b296d9","merge_sha":"a5493c220346b834c479119a0a356717fd6c1c3e"} -->

weihanglo commented:
Valid.
Should either wait or revert.

@bors r-

bors commented:
:sunny: Try build successful - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/4239738297/jobs/7370227844)
Build commit: a5493c220346b834c479119a0a356717fd6c1c3e (`a5493c220346b834c479119a0a356717fd6c1c3e`)
<!-- homu: {"type":"TryBuildCompleted","builders":{"checks-actions":"https://github.com/rust-lang-ci/rust/actions/runs/4239738297/jobs/7370227844"},"merge_sha":"a5493c220346b834c479119a0a356717fd6c1c3e"} -->

bors commented:
:sunny: Try build successful - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/4239738297/jobs/7370227844)
Build commit: a5493c220346b834c479119a0a356717fd6c1c3e (`a5493c220346b834c479119a0a356717fd6c1c3e`)
<!-- homu: {"type":"TryBuildCompleted","builders":{"checks-actions":"https://github.com/rust-lang-ci/rust/actions/runs/4239738297/jobs/7370227844"},"merge_sha":"a5493c220346b834c479119a0a356717fd6c1c3e"} -->

weihanglo committed:
Update cargo

15 commits in 17b3d0de0897e1c6b8ca347bd39f850bb0a5b9f6..9d5b32f503fc099c4064298465add14d4bce11e6
2023-02-17 19:45:09 +0000 to 2023-02-22 23:04:16 +0000

- refactor(job_queue): docs and move types around (rust-lang/cargo#11758)
- Scrub more of the test environment (rust-lang/cargo#11757)
- Make more reads of environment variables go through the `Config` (rust-lang/cargo#11754)
- Revert "Update curl-sys to use libcurl 7.88.1" (rust-lang/cargo#11755)
- use consistent case (rust-lang/cargo#11748)
- Switch some tests from `build` to `check` (rust-lang/cargo#11725)
- Fix typo in sparse-registry warning message (rust-lang/cargo#11753)
- reuse url encoding from `url` crate, don't use separate `percent-encoding` (rust-lang/cargo#11750)
- Read environment variables through `Config` instead of `std::env::var(_os)` (rust-lang/cargo#11727)
- Update curl-sys to use libcurl 7.88.1 (rust-lang/cargo#11749)
- mdman: update pretty_assertions to reduce deps (rust-lang/cargo#11747)
- Cleanup tests (rust-lang/cargo#11745)
- Enhance help texts of position args (rust-lang/cargo#11740)
- Fix typo (rust-lang/cargo#11741)
- Update comment about cargo-ok (rust-lang/cargo#11724)

weihanglo commented:
curl has been reverted.

@bors r+ p=1

bors commented:
:pushpin: Commit 80bea6cfce89500cdc65ca5f3824e668956f8c99 has been approved by `weihanglo`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=weihanglo 80bea6cfce89500cdc65ca5f3824e668956f8c99 -->
<!-- homu: {"type":"Approved","sha":"80bea6cfce89500cdc65ca5f3824e668956f8c99","approver":"weihanglo","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

bors commented:
:hourglass: Testing commit 80bea6cfce89500cdc65ca5f3824e668956f8c99 with merge 8b1dbf728add722d4db894b9b986ec24e1cdb0a1...
<!-- homu: {"type":"BuildStarted","head_sha":"80bea6cfce89500cdc65ca5f3824e668956f8c99","merge_sha":"8b1dbf728add722d4db894b9b986ec24e1cdb0a1"} -->

bors commented:
:sunny: Test successful - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/4249076480/jobs/7390722014)
Approved by: weihanglo
Pushing 8b1dbf728add722d4db894b9b986ec24e1cdb0a1 to master...
<!-- homu: {"type":"BuildCompleted","approved_by":"weihanglo","base_ref":"master","builders":{"checks-actions":"https://github.com/rust-lang-ci/rust/actions/runs/4249076480/jobs/7390722014"},"merge_sha":"8b1dbf728add722d4db894b9b986ec24e1cdb0a1"} -->

bors commented:
:sunny: Test successful - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/4249076480/jobs/7390722014)
Approved by: weihanglo
Pushing 8b1dbf728add722d4db894b9b986ec24e1cdb0a1 to master...
<!-- homu: {"type":"BuildCompleted","approved_by":"weihanglo","base_ref":"master","builders":{"checks-actions":"https://github.com/rust-lang-ci/rust/actions/runs/4249076480/jobs/7390722014"},"merge_sha":"8b1dbf728add722d4db894b9b986ec24e1cdb0a1"} -->

rust-timer commented:
Finished benchmarking commit (8b1dbf728add722d4db894b9b986ec24e1cdb0a1): [comparison URL](https://perf.rust-lang.org/compare.html?start=0978711950b77582e4f8f334f6e9848d48ab7790&end=8b1dbf728add722d4db894b9b986ec24e1cdb0a1&stat=instructions:u).

### Overall result: ❌✅ regressions and improvements - ACTION NEEDED


**Next Steps**: If you can justify the regressions found in this perf run, please indicate this with `@rustbot label: +perf-regression-triaged` along with sufficient written justification. If you cannot justify the regressions please open an issue or create a new PR that fixes the regressions, add a comment linking to the newly created issue or PR, and then add the `perf-regression-triaged` label to this PR.

@rustbot label: +perf-regression
cc @rust-lang/wg-compiler-performance



### [Instruction count](https://perf.rust-lang.org/compare.html?start=0978711950b77582e4f8f334f6e9848d48ab7790&end=8b1dbf728add722d4db894b9b986ec24e1cdb0a1&stat=instructions:u)
This is a highly reliable metric that was used to determine the overall result at the top of this comment.

|                                    | mean  | range          | count |
|:----------------------------------:|:-----:|:--------------:|:-----:|
| Regressions ❌ <br /> (primary)    | -     | -              | 0     |
| Regressions ❌ <br /> (secondary)  | 2.5%  | [2.5%, 2.5%]   | 1     |
| Improvements ✅ <br /> (primary)   | -1.0% | [-1.1%, -0.9%] | 2     |
| Improvements ✅ <br /> (secondary) | -2.2% | [-2.5%, -1.9%] | 6     |
| All ❌✅ (primary)                 | -1.0% | [-1.1%, -0.9%] | 2     |

### [Max RSS (memory usage)](https://perf.rust-lang.org/compare.html?start=0978711950b77582e4f8f334f6e9848d48ab7790&end=8b1dbf728add722d4db894b9b986ec24e1cdb0a1&stat=max-rss)
<details>
<summary>Results</summary>

This is a less reliable metric that may be of interest but was not used to determine the overall result at the top of this comment.

|                                    | mean  | range          | count |
|:----------------------------------:|:-----:|:--------------:|:-----:|
| Regressions ❌ <br /> (primary)    | -     | -              | 0     |
| Regressions ❌ <br /> (secondary)  | -     | -              | 0     |
| Improvements ✅ <br /> (primary)   | -3.1% | [-3.1%, -3.1%] | 1     |
| Improvements ✅ <br /> (secondary) | -2.9% | [-2.9%, -2.9%] | 1     |
| All ❌✅ (primary)                 | -3.1% | [-3.1%, -3.1%] | 1     |
</details>

### [Cycles](https://perf.rust-lang.org/compare.html?start=0978711950b77582e4f8f334f6e9848d48ab7790&end=8b1dbf728add722d4db894b9b986ec24e1cdb0a1&stat=cycles:u)
<details>
<summary>Results</summary>

This is a less reliable metric that may be of interest but was not used to determine the overall result at the top of this comment.

|                                    | mean | range        | count |
|:----------------------------------:|:----:|:------------:|:-----:|
| Regressions ❌ <br /> (primary)    | -    | -            | 0     |
| Regressions ❌ <br /> (secondary)  | 3.6% | [1.7%, 8.7%] | 25    |
| Improvements ✅ <br /> (primary)   | -    | -            | 0     |
| Improvements ✅ <br /> (secondary) | -    | -            | 0     |
| All ❌✅ (primary)                 | -    | -            | 0     |
</details>

nnethercote commented:
Clearly noise, when getting perf changes from a cargo update.

@rustbot label: +perf-regression-triaged
